rllib/agents/ddpg/ddpg.py

```diff
@@ -1,189 +1,296 @@
+                Example:
+                    training_intensity=1000.0
+                    train_batch_size=250 rollout_fragment_length=1
+                    num_workers=1 (or 0) num_envs_per_worker=1
+                    -> natural value = 250 / 1 = 250.0
+                    -> will make sure that replay+train op will be executed 4x as
+                    often as rollout+insert op (4 * 250 = 1000).
+                See: rllib/agents/dqn/dqn.py::calculate_rr_weights for further details.
+        Returns:
+            This updated DDPGConfig object.
+        """
+        super().training(**kwargs)
+
+        if twin_q is not None:
+            self.twin_q = twin_q
+        if policy_delay is not None:
+            self.policy_delay = policy_delay
+        if smooth_target_policy is not None:
+            self.smooth_target_policy = smooth_target_policy
+        if target_noise is not None:
+            self.target_noise = target_noise
+        if target_noise_clip is not None:
+            self.target_noise_clip = target_noise_clip
+        if use_state_preprocessor is not None:
+            self.use_state_preprocessor = use_state_preprocessor
+        if actor_hiddens is not None:
+            self.actor_hiddens = actor_hiddens
+        if actor_hidden_activation is not None:
+            self.actor_hidden_activation = actor_hidden_activation
+        if critic_hiddens is not None:
+            self.critic_hiddens = critic_hiddens
+        if critic_hidden_activation is not None:
+            self.critic_hidden_activation = critic_hidden_activation
+        if n_step is not None:
+            self.n_step = n_step
+        if critic_lr is not None:
+            self.critic_lr = critic_lr
+        if actor_lr is not None:
+            self.actor_lr = actor_lr
+        if target_network_update_freq is not None:
+            self.target_network_update_freq = target_network_update_freq
+        if tau is not None:
+            self.tau = tau
+        if use_huber is not None:
+            self.use_huber = use_huber
+        if huber_threshold is not None:
+            self.huber_threshold = huber_threshold
+        if l2_reg is not None:
+            self.l2_reg = l2_reg
+        if worker_side_prioritization is not None:
+            self.worker_side_prioritization = worker_side_prioritization
+        if training_intensity is not None:
+            self.training_intensity = training_intensity
+
+        return self
+
+
+# Deprecated: Use ray.rllib.agents.ddpg.DDPGConfig instead!
+class _deprecated_default_config(dict):
+    def __init__(self):
+        super().__init__(DDPGConfig().to_dict())
+
+    @Deprecated(
+        old="ray.rllib.agents.ddpg.ddpg::DEFAULT_CONFIG",
+        new="ray.rllib.agents.ddpg.ddpg.DDPGConfig(...)",
+        error=False,
+    )
+    def __getitem__(self, item):
+        return super().__getitem__(item)
+
+
+DEFAULT_CONFIG = _deprecated_default_config()
```

Comment:
You will have to move this all the way down, as it otherwise creates a circular import error (DDPGTrainer not defined yet when you go into the `_deprecated_default_config`'s c'tor to construct the old-style DEFAULT_CONFIG object).

Reply:
oh yeah, of course